### ISSUE TITLE
Tabs component tab index not functional / accessibility issue [SCREENCAPS ATTACHED] 

### ISSUE THREAD

**Issue by Haeven** [radix-ui/primitives#2614]
### Current Behavior

Tab indexing (a core accessibility feature) is currently not functional for the Tabs component, specifically you cannot tab into inactive tabs as depicted in the following videos. I really enjoy using this library and resonate strongly with the values of the creator, so it would be my pleasure to submit a PR for this 😄

[VIDEOS]

https://github.com/radix-ui/primitives/assets/42015067/72445e3c-9d45-4d7a-ac7e-919b14ab7b7c

### Expected behavior

When tab indexing through tabs you should be able to interact with inactive tabs.

### Reproducible example

[CodeSandbox Template](https://codesandbox.io/s/2r30e)

### Suggested solution

I noticed the tabindex attribute being set to -1 programmatically on tab-switch, however if tabindex is left at value 0 this issue does not occur.

I would love to start contributing to the Radix project if possible, the required change to fix this effect is depicted here
 
![Screenshot 2023-12-27 at 6 28 14 PM](https://github.com/radix-ui/primitives/assets/42015067/132e7425-d4f1-45f3-bf35-b04576251685)
![Screenshot 2023-12-27 at 6 28 06 PM](https://github.com/radix-ui/primitives/assets/42015067/bed55730-3134-4dd8-83fb-b30a2cf8c24c)
![Screenshot 2023-12-27 at 6 27 59 PM](https://github.com/radix-ui/primitives/assets/42015067/dd82df0f-84f1-4a09-b8a4-09a3d710d562)

### Additional context

Video showing suggested solution PoC

https://github.com/radix-ui/primitives/assets/42015067/84053086-549c-47af-a17e-ea4312360c38

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |         |         |
| React            | n/a     |         |
| Browser          | Tested 3 | Latest |
| Assistive tech   |         |         |
| Node             | n/a     |         |
| npm/yarn         |         |         |
| Operating System |         |         |

**Comment by Haeven** [comment]
You can use the keyboard to index between inactive tabs
